core/io/stream_peer.h

```diff
@@ -50,13 +50,23 @@ class StreamPeer : public Reference {
 	bool big_endian;
 
 public:
+	enum Status {
+
+		STATUS_CLOSED,
+		STATUS_CONNECTING,
+		STATUS_CONNECTED,
+		STATUS_ERROR,
+	};
+
 	virtual Error put_data(const uint8_t *p_data, int p_bytes) = 0; ///< put a whole chunk of data, blocking until it sent
 	virtual Error put_partial_data(const uint8_t *p_data, int p_bytes, int &r_sent) = 0; ///< put as much data as possible, without blocking.
 
 	virtual Error get_data(uint8_t *p_buffer, int p_bytes) = 0; ///< read p_bytes of data, if p_bytes > available, it will block
 	virtual Error get_partial_data(uint8_t *p_buffer, int p_bytes, int &r_received) = 0; ///< read as much data as p_bytes into buffer, if less was read, return in r_received
 
+	virtual Status get_status() = 0; // Return the Status of this stream.
 	virtual int get_available_bytes() const = 0;
+	virtual Error poll() { return OK; } // Poll the stream looking for change in status and available bytes.
```

Comment:
I know this is meant to be overridden in the specific implementations, but it seems odd to me that we're defaulting to returning OK on a method that doesn't do anything.

Reply:
Well, if the Stream does not need polling and you try to poll it, there should be no error I think.
But we could also use: `ERR_UNAVAILABLE`, or better `ERR_SKIP`

Reply:
To notify the user that the operation was skipped.
Yes, I guess that might work